CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# NOTE: This is NOT a CMakeLists file for OpenDDS itself, OpenDDS must be
+# configured using the configure script. This is one for building all examples
+# and tests that are built using CMake.
```

Comment:
> Can we do without this file?

Possibly, but having this file makes adding new CMake tests and examples much eaiser.

> Maybe the DevGuideExample doesn't need any CMake in it?

No because it's the only realistic CMake example.

> Or root the project somewhere else if add_subdirectory supports .. in relative paths

I would have to look into that.

Reply:
Having two different "root projects" doesn't seem like it would be maintenance issue going forward.  Maybe a bit more setup up front.

Reply:
> Or root the project somewhere else if add_subdirectory supports .. in relative paths

Tried this:

```cmake
add_subdirectory("../../DevGuideExamples/DCPS")
```

and CMake doesn't really seem to like it:

```
CMake Error at CMakeLists.txt:7 (add_subdirectory):
  add_subdirectory not given a binary directory but the given source
  directory "/home/fred/oci/dds/cpp11-cmake/OpenDDS/DevGuideExamples/DCPS" is
  not a subdirectory of "/home/fred/oci/dds/cpp11-cmake/OpenDDS/tests/cmake".
  When specifying an out-of-tree source a binary directory must be explicitly
  specified.
```

As it says I can try to set the binary directory, but it can't be the same as the current one, so I don't really know what to do with it. I can do something like this:

```cmake
add_subdirectory("../../DevGuideExamples/DCPS" "../../DevGuideExamples/DCPS/build")
```

But I'd rather not.

> Having two different "root projects" doesn't seem like it would be maintenance issue going forward.

No, but what I like about this it makes it easy to add more projects that are not in `DevGuideExamples/DCPS` or `tests/cmake`.